src/OSPSuite.Presentation/UICommands/RunMultipleParameterIdentificationsUICommand.cs

```diff
@@ -0,0 +1,30 @@
+﻿using OSPSuite.Core.Domain.ParameterIdentifications;
+using OSPSuite.Core.Domain.Services.ParameterIdentifications;
+using OSPSuite.Core.Extensions;
+using OSPSuite.Core.Services;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace OSPSuite.Presentation.UICommands
+{
+    public class RunMultipleParameterIdentificationsUICommand : ActiveObjectUICommand<IReadOnlyList<ParameterIdentification>>
+   {
+      private readonly IParameterIdentificationRunner _parameterIdentificationRunner;
+
+      public RunMultipleParameterIdentificationsUICommand(
+         IParameterIdentificationRunner parameterIdentificationRunner,
+         IActiveSubjectRetriever activeSubjectRetriever)
+         : base(activeSubjectRetriever)
+      {
+         this._parameterIdentificationRunner = parameterIdentificationRunner;
+      }
+
+      protected override async void PerformExecute()
+      {
+         var tasks = Subject.Select(pi => _parameterIdentificationRunner.SecureAwait(x => x.Run(pi)));
+
+         await Task.WhenAll(tasks);
```

Comment:
@rwmcintosh  That would make little sense to me since they have all been awaited. But maybe it's too early for me

Reply:
Hmm now I'm not sure. We want to run them and possibly show the visual feedback. So sequential run is probably the only way. We can discuss.

Reply:
Maybe I can leave that line but I'm uncertain. As mentioned, the tasks run concurrently, and I'm unsure of the implications if we return control to the surrounding code immediately after initiating the asynchronous tasks. For instance, could this create race conditions?

For example this code in the baseclass 'ObjectUIComannds': 

 public virtual void Execute()
 {
    PerformExecute();
    Subject = null;
 }
 what is the effect of making the Subject null when the tasks are still runninng? 
 
 I can be totally wrong here. Concurrency remains a challenging concept, but these are my current insights.

Reply:
We decided in a meeting that the await-whenall rule could be safely omitted because the calling code is stateless. Also that we  can have the tasks executed concurrently.